app/controllers/admin/petition_details_controller.rb

```diff
@@ -0,0 +1,30 @@
+class Admin::PetitionDetailsController < Admin::AdminController
+  respond_to :html
+  before_action :find_petition
+
+  def show
+    if @petition.in_todo_list?
```

Comment:
We could use a guard instead of an empty `if` statement, so the whole action would end up reading like this:

`raise ActiveRecord::RecordNotFound unless @petition.in_todo_list?`

I think you could extract line this to a separate method and use it in `update` action as well. Currently `update` will accept editing a petition that is not in todo-list, if you tamper the request (probably not a big deal since this is an admin feature).